examples/terraform/hetzner/versions.tf

```diff
@@ -1,4 +1,4 @@
 
 terraform {
-  required_version = ">= 0.12"
+  required_version = ">= 0.13"
```

Comment:
I've successfully provisioned infra using this config and Terraform v0.12. What problems have you encountered with v0.12? Here's the output of `terraform version` for me:

```
Terraform v0.12.23
+ provider.hcloud v1.20.1

Your version of Terraform is out of date! The latest version
is 0.13.0. You can update by downloading from https://www.terraform.io/downloads.html
```

Reply:
If no providers are installed i got following error:

```
$ terraform init                                                                                 

Initializing the backend...

Initializing provider plugins...
- Finding latest version of hashicorp/hcloud...

Error: Failed to install providers

Could not find required providers, but found possible alternatives:

  hashicorp/hcloud -> terraform-providers/hcloud

If these suggestions look correct, upgrade your configuration with the
following command:
    terraform 0.13upgrade .
```

Reply:
Yeah, I was able to provision it with 0.12 successfully too. So let's please not upgrade terraform. Especially given that we use 0.12 in e2e tests.

Reply:
I agree with @kron4eg. Let's keep 0.12, as all other configs and E2E tests are still using 0.12.

Reply:
OK, let's get rid of this `required_version = ">= 0.13"` and this PR is good to merge

Reply:
done!